src/avitab/apps/HeaderApp.cpp

```diff
@@ -100,20 +103,34 @@ void HeaderApp::onBrightnessChange(int brightness) {
     api().setBrightness(brightness / 100.0f);
 }
 
+void HeaderApp::onClockClick(int x, int y, bool pr, bool rel) {
+    if (pr) {
+        stopwatchMode = !stopwatchMode;
+        timerCount = 0;
+        updateClock();
+    }
+}
+
 bool HeaderApp::onTick() {
     updateClock();
     updateFPS();
     return true;
 }
 
 void HeaderApp::updateClock() {
-    std::string time = platform::getLocalTime("%H:%M");
-    if (curTimeString != time) {
-        // to prevent rendering calls each tick
-        curTimeString = time;
-        clockLabel->setText(time);
+    if ((timerCount % TIMER_TICKS_PER_SEC) == 0) {
+        std::ostringstream t;
+        if (stopwatchMode) {
+            unsigned int mins = (timerCount/TIMER_TICKS_PER_SEC) / 60;
+            unsigned int secs = (timerCount/TIMER_TICKS_PER_SEC) % 60;
+            t << std::setfill('0') << std::setw(2) << mins << ":" << std::setw(2) << secs;
+        } else {
+            t << platform::getLocalTime("%H:%M");
+        }
+        clockLabel->setText(t.str());

```

Comment:
I was about to comment that the logic to only call `setText` if the text actually changed was important so that LVGL doesn't redraw the header every 100 ms even though it changes only once a minute, but it seems that I (or you?) added this logic into `setText` at one point, so this is perfectly fine 👍